src/proxy_app/rate_limiter.py

```
@@ -28,6 +44,28 @@ def _parse_reset_time_utc(time_str: str) -> datetime:
     return reset
 
 
+def _resolve_tz(tz_name: str):
+    """Return a tzinfo for the given name; fall back to UTC if unavailable."""
+    if not tz_name or str(tz_name).upper() == "UTC":
+        return timezone.utc
+    if _HAS_ZONEINFO:
+        try:
+            return ZoneInfo(tz_name)
+        except Exception as e:  # pragma: no cover - defensive
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)